src/dotnet/APIView/APIViewWeb/Managers/NotificationManager.cs

```diff
@@ -4,52 +4,61 @@
 using APIView.Identity;
 using APIViewWeb.Models;
 using Microsoft.Extensions.Configuration;
-using SendGrid;
-using SendGrid.Helpers.Mail;
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Net;
 using System.Security.Claims;
 using System.Text;
 using System.Threading.Tasks;
-using Microsoft.TeamFoundation.Common;
 using APIViewWeb.Repositories;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.ApplicationInsights;
+using System.Net.Http;
+using System.Net.Http.Json;
 
 namespace APIViewWeb.Managers
 {
     public class NotificationManager : INotificationManager
     {
-        private readonly string _endpoint;
+        private readonly string _apiviewEndpoint;
         private readonly ICosmosReviewRepository _reviewRepository;
         private readonly ICosmosUserProfileRepository _userProfileRepository;
-        private readonly ISendGridClient _sendGridClient;
+        private readonly IConfiguration _configuration;
+        private readonly string _testEmailToAddress;
+        private readonly string _emailSenderServiceUrl;
 
         private const string ENDPOINT_SETTING = "Endpoint";
-        private const string SENDGRID_KEY_SETTING = "SendGrid:Key";
-        private const string FROM_ADDRESS = "apiview-noreply@microsoft.com";
-        private const string REPLY_TO_HEADER = "In-Reply-To";
-        private const string REFERENCES_HEADER = "References";
 
-        public NotificationManager(IConfiguration configuration, ICosmosReviewRepository reviewRepository,
-        ICosmosUserProfileRepository userProfileRepository, ISendGridClient sendGridClient = null)
+        static TelemetryClient _telemetryClient = new(TelemetryConfiguration.CreateDefault());
+
+        public NotificationManager(IConfiguration configuration,
+            ICosmosReviewRepository reviewRepository,
+            ICosmosUserProfileRepository userProfileRepository)
         {
-            _sendGridClient = sendGridClient ?? new SendGridClient(configuration[SENDGRID_KEY_SETTING]);
-            _endpoint = configuration.GetValue<string>(ENDPOINT_SETTING);
+            _apiviewEndpoint = configuration.GetValue<string>(ENDPOINT_SETTING);
             _reviewRepository = reviewRepository;
             _userProfileRepository = userProfileRepository;
+            _configuration = configuration;
+            _testEmailToAddress = configuration["apiview-email-test-address"] ?? "";
```

Comment:
What does test address mean in this context?

Reply:
This is to enable testing on local and on even deployed instances if required to a specific email address instead of using actual email address on test instance.